File: LeetCode_Record/2287. 重排字符形成目标字符串.cpp
#include <string>
using namespace std;

class Solution {
public:
    int rearrangeCharacters(string s, string target) {
        int chr[26] = {0};
        int chrt[26] = {0};
        for(char& i : s){
            chr[i - 'a']++;
        }
        for(char& i : target){
            chrt[i - 'a']++;
        }
        int ans = INT_MAX;
        for(int i = 0; i < 26; i++){
            if(chrt[i]){
                if(chr[i] < chrt[i])
                    return 0;
                ans = min(ans, chr[i] / chrt[i]);
            }
        }
        return ans;
    }
};

File: LeetCode_Record/2287. 重排字符形成目标字符串.java
import java.lang.String;
import java.security.KeyStore.Entry;
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int rearrangeCharacters(String s, String target) {
        HashMap<Integer, Integer> chr = new HashMap<>();
        HashMap<Integer, Integer> chrt = new HashMap<>();
        s.chars().forEach(v -> chr.put(v, chr.getOrDefault(v, 0) + 1));
        target.chars().forEach(v -> chrt.put(v, chrt.getOrDefault(v, 0) + 1));
        int ans = Integer.MAX_VALUE;
        for(Map.Entry<Integer, Integer> e : chrt.entrySet()){
            if(chr.getOrDefault(e.getKey(), 0) < chrt.get(e.getKey()))
                return 0;
            ans = Integer.min(ans, chr.get(e.getKey())/chrt.get(e.getKey()));
        }
        return ans;
    }
}